lib/server-jobs.js

```diff
@@ -476,7 +483,27 @@ module.exports.getJobSequence = function (job_sequence_id, course_id, callback)
   };
   sqldb.queryOneRow(sql.select_job_sequence_with_course_id_as_json, params, function (err, result) {
     if (ERR(err, callback)) return;
-    callback(null, result.rows[0]);
+
+    const jobSequence = result.rows[0];
+
+    // Generate a token to authorize websocket requests for this job sequence.
+    const jobSequenceTokenData = {
+      jobSequenceId: job_sequence_id.toString(),
+    };
+    jobSequence.token = csrf.generateToken(jobSequenceTokenData, config.secretKey);
+
+    (jobSequence.jobs ?? []).forEach((job) => {
+      // Also generate a token for each job.
+      const jobTokenData = { jobId: job.id.toString() };
+      job.token = csrf.generateToken(jobTokenData, config.secretKey);
+
+      if (job.output) {
+        const ansiup = new AnsiUp();
+        job.output = ansiup.ansi_to_html(job.output);
```

Comment:
I think these lines should not have been copied from `getJobSequenceWithFormattedOutput()` to this location in `getJobSequence()`?

Reply:
Indeed! Removed in 7afbe51.